lib/core/src/system/web/dom/html/css3d_renderer.rs

```diff
@@ -0,0 +1,269 @@
+        back_dom.set_property_or_panic("z-index", "-1");
+        front_camera.set_property_or_panic("width", "100%");
+        front_camera.set_property_or_panic("height", "100%");
+        front_camera.set_property_or_panic("transform-style", "preserve-3d");
+        back_camera.set_property_or_panic("width", "100%");
+        back_camera.set_property_or_panic("height", "100%");
+        back_camera.set_property_or_panic("transform-style", "preserve-3d");
+
+        container.dom.append_or_panic(&front_dom);
+        container.dom.append_or_panic(&back_dom);
+        front_dom.append_or_panic(&front_camera);
+        back_dom.append_or_panic(&back_camera);
+
+        let data             = Css3dRendererData::new(front_dom, back_dom, front_camera, back_camera);
+        let data             = Rc::new(data);
+        let mut renderer     = Self {container,data,logger};
+
+        renderer.init_listeners();
+        Ok(renderer)
+    }
+
+    /// Creates a Css3dRenderer.
+    pub fn new<L:Into<Logger>>(logger:L, dom_id: &str) -> Result<Self> {
+        Self::from_element(logger,dyn_into(get_element_by_id(dom_id)?)?)
+    }
+
+    pub(crate) fn logger(&self) -> Logger {
+        self.logger.clone()
+    }
+
+    fn init_listeners(&mut self) {
+        let dimensions = self.dimensions();
+        self.set_dimensions(dimensions);
+        let data = self.data.clone();
+        self.add_resize_callback(move |dimensions:&Vector2<f32>| {
+            data.set_dimensions(*dimensions);
+        });
+    }
+
+    /// Creates a new instance of Css3dObject and adds it to parent.
+    pub(super) fn new_instance
+    <S:AsRef<str>>(&self, dom_name:S, parent:DisplayObjectData) -> Result<Css3dObject> {
+        let front_camera = self.data.front_camera.clone();
+        let back_camera  = self.data.back_camera.clone();
+        let object = Css3dObject::new(self.logger.sub("object"),dom_name,front_camera,back_camera);
+        object.as_ref().map(|object| {
+            parent.add_child(object);
+        }).ok();
+        object
+    }
+
+    /// Renders `Camera`'s point of view.
+    pub fn render(&self, camera: &Camera2d) {
+        let trans_cam  = camera.transform().matrix().try_inverse();
+        let trans_cam  = trans_cam.expect("Camera's matrix is not invertible.");
+        let trans_cam  = trans_cam.map(eps);
+        let trans_cam  = invert_y(trans_cam);
+        let half_dim   = self.container.dimensions() / 2.0;
+        let fovy_slope = camera.half_fovy_slope();
+        let near       = half_dim.y / fovy_slope;
+
+        match camera.projection() {
+            Projection::Perspective{..} => {
+                js::setup_perspective(&self.data.front_dom, &near.into());
+                js::setup_perspective(&self.data.back_dom, &near.into());
+                setup_camera_perspective(
+                    &self.data.front_camera,
+                    near,
+                    &trans_cam
+                );
```

Comment:
why not single line?

Reply:
'cause it didn't before :P I just fixed it